### ISSUE TITLE
Add "fog" to environment description

### ISSUE THREAD

**Issue by hschoen** [OpenSimulationInterface/open-simulation-interface#54]
In osi-environment.proto precipitation is defined. However, fog cannot be described with mm/h. I suggest to add fog like this:

    /// Fog
    optional Fog fog = 5;

    /// Definition of fog states. Values need to be revised! See https://de.wikipedia.org/wiki/Nebel#Tr.C3.B6pfchengr.C3.B6.C3.9Fe_und_Nebeldichte Use visibility instead? Values for radar attenuation required!
    {
        /// Light intensity fog, when density is < 0.02 g/m³
        PRECIPITATION_FOG_LIGHT = 0;
  
        /// Medium intensity fog, when density is < 0.05 g/m³
        PRECIPITATION_FOG_MEDIUM = 1;
  
        /// High intensity fog, when density is < 0.1 g/m³
        PRECIPITATION_FOG_HIGH = 2;
  
        /// Violent intensity fog, when density is > 0.2 g/m³
        PRECIPITATION_FOG_VIOLENT = 3;  
    }

**Comment by CarloVanDriestenBMW** [comment]
Yes, that would be OK for me, we should talk about that at the workshop and find solid references for both :)

**Comment by CarloVanDriestenBMW** [comment]
The enum here is not defined correctly:

```
    /// Definition of precipitation states. See https://en.wikipedia.org/wiki/Rain#Measurement
    enum Precipitation
    {
        /// Light intensity rain, when the precipitation rate is < 2.5 mm per hour
        PRECIPITATION_RAIN_LIGHT = 0;
  
        /// Medium intensity rain, when the precipitation rate is between 2.5 mm and 10 mm per hour
        PRECIPITATION_RAIN_MEDIUM = 1;
  
        /// High intensity rain, when the precipitation rate is between 10 mm and 50 mm per hour
        PRECIPITATION_RAIN_HIGH = 2;
  
        /// Violent intensity rain, when the precipitation rate is > 50 mm per hour
        PRECIPITATION_RAIN_VIOLENT = 3;  
} 
```

@AhmedMzid and @PhilippRoethigBMW please find better definitions for fog and rain according to DIN standards e.g.

**Comment by PhilippRoethigBMW** [comment]
@CarloVanDriestenBMW what about a rain and fog state similar to this one:

```cmd
    /// No rain, when the precipitation rate is < 0.1 mm per hour (values are not final)
    PRECIPITATION_RAIN_NONE = 2;
```

**Comment by CarloVanDriestenBMW** [comment]
Do you have any source for your proposal?

**Comment by hschoen** [comment]
@CarloVanDriestenBMW Maybe we should define fog by visibility which is not so well defined but more practical. Here is a source in which attenuation due to fog is related to visibility: 
Meteorological Radar, Serafin in Radar Handbook (2nd edition), Skolnik, 1990; p. 23.10-11
However, the shortest wavelength in this table is 1.25 cm (77 GHz -> 0.4 cm) (could be extrapolated).

**Comment by PhilippRoethigBMW** [comment]
I do also think defining fog by visibility would be a good idea.

@hschoen have a look at the changes made in pull request #59 
Do they address your issue adequately?

See this [overview ](https://github.com/OpenSimulationInterface/open-simulation-interface/files/882747/Environment_classification.pdf)for alternative values from other sources.
The values I used are marked yellow.

**Comment by CarloVanDriestenBMW** [comment]
solved with pull request #59
